src/Sylius/Bundle/ProductBundle/Resources/config/services.xml

```diff
@@ -44,6 +47,12 @@
             <argument type="service" id="sylius.manager.product" />
             <tag name="kernel.event_listener" event="sylius.product_archetype.pre_update" method="onArchetypeUpdate" />
         </service>
+
+        <service id="sylius.form.type.product_association" class="%sylius.form.type.product_association.class%">
```

Comment:
Why??? This form should be generated by SyliusAssociationBundle. Just like it is done for attributes and options, etc.